### PR TITLE
Add basic block/machine framework

### PR DESCRIPTION
**Pull request by Lvgraef**
Adds the base blocks and basic machine framework.

### CODE REVIEW

File: src/main/kotlin/earth/terrarium/techarium/common/blocks/base/DirectionalDeployableMachine.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        registerDefaultState(defaultBlockState()
+            .setValue(FACING, Direction.NORTH))
+    }
+
+    override fun createBlockStateDefinition(builder: StateDefinition.Builder<Block, BlockState>) {
+        super.createBlockStateDefinition(builder)
+        builder.add(FACING)
+    }
+
+    override fun rotate(state: BlockState, level: LevelAccessor, pos: BlockPos, direction: Rotation): BlockState {
```
Comment (MsRandom) [inline src/main/kotlin/earth/terrarium/techarium/common/blocks/base/DirectionalDeployableMachine.kt]: Maybe a nitpick, but these can be expression functions

Comment (Lvgraef) [reply src/main/kotlin/earth/terrarium/techarium/common/blocks/base/DirectionalDeployableMachine.kt]: you are right I will change them

File: src/main/kotlin/earth/terrarium/techarium/common/util/Extensions.kt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package earth.terrarium.techarium.common.util
```
Comment (ThatGravyBoat) [inline src/main/kotlin/earth/terrarium/techarium/common/util/Extensions.kt]: This should be in a package called `utils` as plural is used for all other packages.
This should also not be called `Extensions` something like BlockUtils would be better as we should not be putting all extensions in one file.

File: src/main/kotlin/earth/terrarium/techarium/common/registries/ModBlocks.kt [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import earth.terrarium.techarium.common.blocks.machines.deploying.BasicDeployChildBlock
+import net.minecraft.core.registries.BuiltInRegistries
+import net.minecraft.world.level.block.Block
+import net.minecraft.world.level.block.state.BlockBehaviour
+import net.minecraft.world.level.material.PushReaction
+
+object ModBlocks {
+    val registry: ResourcefulRegistry<Block> = ResourcefulRegistries.create(BuiltInRegistries.BLOCK, TechariumConstants.MOD_ID)
+
+    val basicDeployChildBlock: BasicDeployChildBlock by registry.register("basic_deploy_child_block") {BasicDeployChildBlock(BlockBehaviour.Properties.of().requiresCorrectToolForDrops().noOcclusion().pushReaction(PushReaction.BLOCK)) }
```
Comment (ThatGravyBoat) [inline src/main/kotlin/earth/terrarium/techarium/common/registries/ModBlocks.kt]: please format this so its on its own line ie.
```
register {
blockCreatioN()
}

File: src/main/kotlin/earth/terrarium/techarium/common/blocks/entities/machines/BasicDeployChildBlockEntity.kt [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package earth.terrarium.techarium.common.blocks.entities.machines
+
+import net.minecraft.core.BlockPos
+import net.minecraft.world.level.block.entity.BlockEntity
+import net.minecraft.world.level.block.entity.BlockEntityType
+import net.minecraft.world.level.block.state.BlockState
+
+class BasicDeployChildBlockEntity(type: BlockEntityType<*>, pos: BlockPos, blockState: BlockState): BlockEntity(type, pos,
```
Comment (ThatGravyBoat) [inline src/main/kotlin/earth/terrarium/techarium/common/blocks/entities/machines/BasicDeployChildBlockEntity.kt]: please format this

File: src/main/kotlin/earth/terrarium/techarium/common/registries/ModBlockEntityTypes.kt [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import com.teamresourceful.resourcefullibkt.common.getValue
+import earth.terrarium.techarium.common.TechariumConstants
+import earth.terrarium.techarium.common.blocks.entities.machines.BasicDeployChildBlockEntity
+import net.minecraft.core.registries.BuiltInRegistries
+import net.minecraft.world.level.block.entity.BlockEntityType
+
+object ModBlockEntityTypes {
+    val registry: ResourcefulRegistry<BlockEntityType<*>> = ResourcefulRegistries.create(BuiltInRegistries.BLOCK_ENTITY_TYPE, TechariumConstants.MOD_ID)
+
+    val basicDeployChildBlockEntity: BlockEntityType<BasicDeployChildBlockEntity> by registry.register("basic_deploy_child_blockentity") {BlockEntityType.Builder.of({pos, state -> BasicDeployChildBlockEntity(basicDeployChildBlockEntity, pos, state)
```
Comment (ThatGravyBoat) [inline src/main/kotlin/earth/terrarium/techarium/common/registries/ModBlockEntityTypes.kt]: same as ModBlocks, format

**Review by ThatGravyBoat — CHANGES_REQUESTED** [review]

**Review by ThatGravyBoat — APPROVED** [review]

**Pull request merged**